File: Компьютерный практикум по моделированию/Лабораторные работы/Лабораторная работа №4/Task2_MethodYakobi.py
import numpy as np

# Функция для выполнения одной итерации метода Якоби
def jacobi_iteration(x, y, z):
    new_x = (-1 - 2*z) / 2
    new_y = (2 - 5*x + 5*z) / -2
    new_z = (0 - 2*x - 2*y) / 5
    return new_x, new_y, new_z

# Начальные значения для метода Якоби
x, y, z = 0, 0, 0

# Количество итераций
iterations = 10

# Итерационный процесс метода Якоби
for i in range(iterations):
    x, y, z = jacobi_iteration(x, y, z)
    
    # Вывод текущей итерации
    print(f"Итерация {i+1}:")
    print(f"x = {x}, y = {y}, z = {z}")

# Вывод финального решения
print("Решение методом Якоби:")
print(f"x = {x}, y = {y}, z = {z}")

File: Компьютерный практикум по моделированию/Лабораторные работы/Лабораторная работа №4/Task2_MethodYakobi.cpp
#include <iostream>
#include <cmath>

using namespace std;

// Функция для вычисления следующей итерации метода Якоби
void jacobiIteration(float& x, float& y, float& z) {
    float newX = (-1 - 2*z) / 2;
    float newY = (2 - 5*x + 5*z) / -2;
    float newZ = (0 - 2*x - 2*y) / 5;
    
    x = newX;
    y = newY;
    z = newZ;
}

int main() {
    // Начальные значения для метода Якоби
    float x = 0, y = 0, z = 0;
    
    // Количество итераций
    int iterations = 10;
    
    // Итерационный процесс метода Якоби
    for (int i = 0; i < iterations; ++i) {
        jacobiIteration(x, y, z);
        
        // Вывод текущей итерации
        cout << "Итерация " << i+1 << ": " << endl;
        cout << "x = " << x << ", y = " << y << ", z = " << z << endl;
    }
    
    // Вывод финального решения
    cout << "Решение методом Якоби:" << endl;
    cout << "x = " << x << ", y = " << y << ", z = " << z << endl;
    
    return 0;
}



